substrate/frame/scheduler/src/lib.rs

```diff
@@ -292,6 +292,13 @@ pub mod pallet {
 
 		/// The preimage provider with which we look up call hashes to get the call.
 		type Preimages: QueryPreimage<H = Self::Hashing> + StorePreimage;
+
+		/// Provider for the block number. Normally this is the `frame_system` pallet.
+		type BlockNumberProvider: BlockNumberProvider<BlockNumber = BlockNumberFor<Self>>;
```

Comment:
same https://github.com/paritytech/polkadot-sdk/pull/5723#discussion_r1827642556

Reply:
Good point. We should probably update in https://github.com/paritytech/polkadot-sdk/pull/3970 as well. Thoughts @shawntabrizi  @muharem ?

Reply:
I agree, no reason to force the same type